internal/langserver/handlers/initialize.go

```diff
@@ -248,14 +247,33 @@ func (svc *service) setupWalker(ctx context.Context, params lsp.InitializeParams
 		return err
 	}
 
-	var excludeModulePaths []string
-	for _, rawPath := range options.ExcludeModulePaths {
+	if len(options.XLegacyModulePaths) != 0 {
+		jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
+			Type: lsp.Warning,
+			Message: fmt.Sprintf("rootModulePaths (%q) is deprecated (no-op), add a folder to workspace "+
+				"instead if you'd like it to be indexed", options.XLegacyModulePaths),
+		})
+	}
+	if len(options.XLegacyExcludeModulePaths) != 0 {
+		jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
+			Type: lsp.Warning,
+			Message: fmt.Sprintf("excludeModulePaths (%q) is deprecated (no-op), use ignorePaths instead",
+				options.XLegacyExcludeModulePaths),
+		})
+	}
+
+	var ignoredPaths []string
+	for _, rawPath := range options.IgnorePaths {
 		modPath, err := resolvePath(root.Path(), rawPath)
 		if err != nil {
-			svc.logger.Printf("Ignoring excluded module path %s: %s", rawPath, err)
+			jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
```

Comment:
Do we want to show a message to the user in this case? For example, if a user switches between many projects, the paths might only apply to some. VS Code has workspace-specific settings for this, but users might not use them or use an entirely different editor.

Reply:
👍🏼 on the no message from server in this case.

Reply:
I filed https://github.com/hashicorp/terraform-ls/issues/1008 but after re-reading the code here I realized the warning would not be sent just because a path doesn't exist - only if the URI is invalid - so that seems like a good reason to continue raising warning to me?